runtime/Go/antlr/utils.go

```diff
@@ -110,48 +126,47 @@ func standardHashFunction(a interface{}) string {
 type Hasher interface {
 	Hash() string
 }
-
-func hashCode(s string) string {
-	h := fnv.New32a()
-	h.Write([]byte((s)))
-	return fmt.Sprint(h.Sum32())
+type HashCodeer interface {
+	HashCode() int
 }
 
 func (s *Set) length() int {
-	return len(s.data)
+	return len(s.setData)
 }
 
 func (s *Set) add(value interface{}) interface{} {
 
-	hash := s.hashFunction(value)
-	key := "hash_" + hashCode(hash)
+	key := s.hashcodeFunction(value)
 
-	values := s.data[key]
+	values := s.setData[key]
 
-	if s.data[key] != nil {
+	if s.setData[key] != nil {
 
 		for i := 0; i < len(values); i++ {
 			if s.equalsFunction(value, values[i]) {
 				return values[i]
 			}
 		}
 
-		s.data[key] = append(s.data[key], value)
+		s.setData[key] = append(s.setData[key], value)
 		return value
 	}
 
-	s.data[key] = []interface{}{value}
+	v := make([]interface{}, 1, 10)
```

Comment:
This isn't a cap,  slice = append( slice, element) with expand the slice as needed.
It was an attempt at a performance optimisation. There are so few clashes in the lexer being benchmarked that it made no difference.